agent/consul/internal_endpoint.go

```diff
@@ -69,18 +76,67 @@ func (m *Internal) NodeDump(args *structs.DCSpecificRequest,
 		&args.QueryOptions,
 		&reply.QueryMeta,
 		func(ws memdb.WatchSet, state *state.Store) error {
-			index, dump, err := state.NodeDump(ws, &args.EnterpriseMeta, args.PeerName)
-			if err != nil {
-				return err
+			peerings := make([]string, 0)
+			peerings = append(peerings, "") // this represents a local fetch
+
+			// if a PeerName is passed, then it's we only get the NodeDump for that peer
+			// otherwise, we get all the local nodes and all the nodes for every peering
+			if args.PeerName != "" {
```

Comment:
We could avoid supporting this use-case of dumping nodes from a peer for now, since we don't have a need for it. Could return an error if a PeerName is specified before resolving the ACL token above.

Reply:
I coded in a short circuit for when `args.PeerName` is passed in at all per our chat.